### DISCUSSION TITLE
Trouble replicating the documentation for Global State/React Context

### DISCUSSION THREAD

**Discussion by mcastrigno** [statelyai/xstate#2924] [Q&A] [answered]
I am attempting to create a simple example to follow the documentation here:
[](https://xstate.js.org/docs/recipes/react.html#global-state-react-context)

I started with create-react-app and added XState. I am attempting to see would be apple to have two separate components where one component could cause a state change and result in the re-render of another component. I trying to do this I want use the global context as described in the documentation. When I attempt to do this however, I get a runtime error relate to useSelector.
( Uncaught TypeError: Cannot use 'in' operator to search for 'state' in undefined )
line 20 in MattsMachineControl.jsx

I am attempting to have the machine in one file, the global state provider in another and the consuming component is a third file as this is how it appears the documentation is written.

My repo is here:
https://github.com/mcastrigno/XState-React

Any help or comments greatly appreciated for new learner. Thank you!

**Comment by mcastrigno** [comment, marked as answer]
The answer here is that the app needs to be wrapped by the provider. Taken verbatim, the example in the XState documentation does not show this. Once I did this, the runtime error was resolved.

**Reply by davidkpiano** [reply]
I'd be happy to accept a PR to the documentation with the missing code added!

**Reply by mcastrigno** [reply]
PR submitted, Thank you.
